Assets/MRTK/SDK/Experimental/Drawstring/Scripts/ElasticHandleLogic.cs

```diff
@@ -0,0 +1,111 @@
+                                    float leftHandleVelocity = 0.0f, float rightHandleVelocity = 0.0f)
+        {
+            isSetup = true;
+            leftInitialPosition = leftHandleStart;
+            rightInitialPosition = rightHandleStart;
+            elasticSystem = new LinearElasticSystem((leftHandleStart - rightHandleStart).magnitude, 0.0f, extentInfo, elasticProperties);
+        }
+
+        /// <summary>
+        /// Update the internal state of the damped harmonic oscillator, given the left and right pointer positions.
+        /// Note, both left and right input positions are nullable; set the pointer position to null
+        /// if only one of the handles is being interacted with. Returns the calculated distance between the
+        /// handles.
+        /// 
+        /// </summary>
+        /// <param name="leftPointer">World position of the pointer manipulating the left handle.</param>
+        /// <param name="rightPointer">World position of the pointer manipulating the right handle.</param>
+        /// <param name="deltaTime">Amount of time to simulate this step.</param>
+        /// <param name="normalizedAxis">World axis of the two-handle system.</param>
+        public virtual float Update(Vector3? leftPointer, Vector3? rightPointer, float deltaTime, Vector3? normalizedAxis = null)
+        {
+            // If we have not been Setup() yet
+            if (!isSetup) { return 0.1f; }
+
+            var handleAxis = normalizedAxis ?? (rightInitialPosition - leftInitialPosition).normalized;
+
+            var handDistance = elasticSystem.GetCurrentValue();
+            if (leftPointer.HasValue && rightPointer.HasValue)
+            {
+                // If we have both pointers, we simply project the vector between the
+                // two hands onto the handle axis vector.
+                handDistance = Vector3.Dot(rightPointer.Value - leftPointer.Value, handleAxis);
+            } else if (!leftPointer.HasValue && rightPointer.HasValue)
+            {
+                // If we only have a right pointer, calculate the hand distance
+                // as twice the distance of the right pointer from the center.
+                handDistance = 2.0f * Vector3.Dot(rightPointer.Value - (leftInitialPosition + rightInitialPosition) / 2.0f, handleAxis);
+            } else if (leftPointer.HasValue && !rightPointer.HasValue)
+            {
+                // If we only have a left pointer, calculate the hand distance
+                // as twice the distance of the left pointer from the center.
+                handDistance = 2.0f * -Vector3.Dot(leftPointer.Value - (leftInitialPosition + rightInitialPosition) / 2.0f, handleAxis);
+            }
+
+            return elasticSystem.ComputeIteration(handDistance, deltaTime);
+        }
+
+        public virtual Vector3 GetCenterPosition(Vector3? leftPointer, Vector3? rightPointer)
```

Comment:
Please add summary comment to this function. thx!

Reply:
Done!